### PR TITLE
option for custom dns resolver

### PR DESCRIPTION
**Pull request by vyzo**

### CODE REVIEW

**Review by Stebalien — COMMENTED** [review]
Don't we actually need to _use_ the resolver somewhere?

**Comment by vyzo** [comment]
the code is already there AFAICT, we just needed to add it to the basic host opts in config. Did I miss that?

File: config/config.go [hunk 1]
```diff
@@ -186,11 +189,12 @@ func (cfg *Config) NewNode(ctx context.Context) (host.Host, error) {
...
-		AddrsFactory: cfg.AddrsFactory,
-		NATManager:   cfg.NATManager,
-		EnablePing:   !cfg.DisablePing,
-		UserAgent:    cfg.UserAgent,
+		ConnManager:       cfg.ConnManager,
+		AddrsFactory:      cfg.AddrsFactory,
+		NATManager:        cfg.NATManager,
+		EnablePing:        !cfg.DisablePing,
+		UserAgent:         cfg.UserAgent,
+		MultiaddrResolver: cfg.MultiaddrResolver,
```
Comment (vyzo) [inline config/config.go:197]: @stebalien we use it here

**Review by Stebalien — APPROVED** [review]
I guess we just never wired this up...

**Pull request merged**